app/src/main/java/com/codingblocks/cbonlineapp/mycourse/MyCourseViewModel.kt

```diff
@@ -113,6 +117,39 @@ class MyCourseViewModel(
         }
     }
 
+    fun fetchExtensions(productId: String?) {
+        runIO {
+            when (val response = productId?.let { repo.getExtensions(Integer.parseInt(it)) }) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    this.enqueue(retrofitCallback { _, response ->
```

Comment:
again what are you trying to achieve here I'm not getting it you should be focusing on understanding the code first rather just completing the work

Reply:
Again the same answer I would give here, being a network request am enquing with the callback to use the response. Let me know if I should use `awaitResponse().body()` directly without enquing